src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if NET8_0_OR_GREATER
```

Comment:
Why limit the entire class?

Reply:
Many parts of the class will rely on SqlDataSource (which extends DbDataSource), so the class will not be functional on .NET lower than 8.